compiler/rustc_expand/src/base.rs

```diff
@@ -347,6 +347,7 @@ pub trait TTMacroExpander {
         ecx: &'cx mut ExtCtxt<'_>,
         span: Span,
         input: TokenStream,
+        nearest_parent: NodeId,
```

Comment:
It shouldn't be necessary to pass `nearest_parent` through all these layers, it's should be already available through `ecx.resolver` at the point of use.

Reply:
Also, I'd prefer to keep using the `lint_node_id` terminology given its approximate nature, the ID is not guaranteed to be the nearest parent.